clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -3349,6 +3357,12 @@ check_kernel_syscall_trace(void)
                 { gen_data(TID_A, false, 42, 8), nullptr },
                 { gen_data(TID_A, false, 42, 8), nullptr },
                 { gen_data(TID_A, false, 42, 8), nullptr },
+                { gen_instr(TID_A), xsaveopt },
+                // Multiple writes and a read. Acceptable because of the prior xsaveopt.
+                { gen_data(TID_A, false, 42, 8), nullptr },
+                { gen_data(TID_A, false, 42, 8), nullptr },
+                { gen_data(TID_A, true, 42, 8), nullptr },
```

Comment:
> Is this what QEMU produces, several stores and then a load and then a store?

Yes. Though the count is much higher (see below). I figured just the interleaved 4 are enough for the test. I'l make the change to make the addresses differnet.

```
       46109       27589:           1 ifetch       4 byte(s) @ 0xffffffff81008257 48 0f ae 37          xsaveopt64 %edx %eax -> (%rdi)
       46110       27589:           1 write        2 byte(s) @ 0xffff8800b904afc0 by PC 0xffffffff81008257
       46111       27589:           1 write        2 byte(s) @ 0xffff8800b904afc2 by PC 0xffffffff81008257
       46112       27589:           1 write        2 byte(s) @ 0xffff8800b904afc4 by PC 0xffffffff81008257
       46113       27589:           1 write        8 byte(s) @ 0xffff8800b904afc8 by PC 0xffffffff81008257
       46114       27589:           1 write        8 byte(s) @ 0xffff8800b904afd0 by PC 0xffffffff81008257
       46115       27589:           1 write        8 byte(s) @ 0xffff8800b904afe0 by PC 0xffffffff81008257
       46116       27589:           1 write        2 byte(s) @ 0xffff8800b904afe8 by PC 0xffffffff81008257
       46117       27589:           1 write        8 byte(s) @ 0xffff8800b904aff0 by PC 0xffffffff81008257
       46118       27589:           1 write        2 byte(s) @ 0xffff8800b904aff8 by PC 0xffffffff81008257
       46119       27589:           1 read         8 byte(s) @ 0x00000000b97f8880 by PC 0xffffffff81008257
       46120       27589:           1 read         8 byte(s) @ 0x0000000003512010 by PC 0xffffffff81008257
       46121       27589:           1 read         8 byte(s) @ 0x0000000003515e40 by PC 0xffffffff81008257
       46122       27589:           1 write        8 byte(s) @ 0xffff8800b904b000 by PC 0xffffffff81008257
       46123       27589:           1 write        2 byte(s) @ 0xffff8800b904b008 by PC 0xffffffff81008257
       46124       27589:           1 write        8 byte(s) @ 0xffff8800b904b010 by PC 0xffffffff81008257
       46125       27589:           1 write        2 byte(s) @ 0xffff8800b904b018 by PC 0xffffffff81008257
       46126       27589:           1 write        8 byte(s) @ 0xffff8800b904b020 by PC 0xffffffff81008257
       46127       27589:           1 write        2 byte(s) @ 0xffff8800b904b028 by PC 0xffffffff81008257
       46128       27589:           1 write        8 byte(s) @ 0xffff8800b904b030 by PC 0xffffffff81008257
       46129       27589:           1 write        2 byte(s) @ 0xffff8800b904b038 by PC 0xffffffff81008257
       46130       27589:           1 write        8 byte(s) @ 0xffff8800b904b040 by PC 0xffffffff81008257
       46131       27589:           1 write        2 byte(s) @ 0xffff8800b904b048 by PC 0xffffffff81008257
       46132       27589:           1 write        8 byte(s) @ 0xffff8800b904b050 by PC 0xffffffff81008257
       46133       27589:           1 write        2 byte(s) @ 0xffff8800b904b058 by PC 0xffffffff81008257
       46134       27589:           1 write        4 byte(s) @ 0xffff8800b904afd8 by PC 0xffffffff81008257
       46135       27589:           1 write        4 byte(s) @ 0xffff8800b904afdc by PC 0xffffffff81008257
       46136       27589:           1 write        8 byte(s) @ 0xffff8800b904b060 by PC 0xffffffff81008257
       46137       27589:           1 write        8 byte(s) @ 0xffff8800b904b068 by PC 0xffffffff81008257
       46138       27589:           1 write        8 byte(s) @ 0xffff8800b904b070 by PC 0xffffffff81008257
       46139       27589:           1 write        8 byte(s) @ 0xffff8800b904b078 by PC 0xffffffff81008257
       46140       27589:           1 write        8 byte(s) @ 0xffff8800b904b080 by PC 0xffffffff81008257
       46141       27589:           1 write        8 byte(s) @ 0xffff8800b904b088 by PC 0xffffffff81008257
       46142       27589:           1 write        8 byte(s) @ 0xffff8800b904b090 by PC 0xffffffff81008257
       46143       27589:           1 write        8 byte(s) @ 0xffff8800b904b098 by PC 0xffffffff81008257
       46144       27589:           1 write        8 byte(s) @ 0xffff8800b904b0a0 by PC 0xffffffff81008257
       46145       27589:           1 write        8 byte(s) @ 0xffff8800b904b0a8 by PC 0xffffffff81008257
       46146       27589:           1 write        8 byte(s) @ 0xffff8800b904b0b0 by PC 0xffffffff81008257
       46147       27589:           1 write        8 byte(s) @ 0xffff8800b904b0b8 by PC 0xffffffff81008257
       46148       27589:           1 write        8 byte(s) @ 0xffff8800b904b0c0 by PC 0xffffffff81008257
       46149       27589:           1 write        8 byte(s) @ 0xffff8800b904b0c8 by PC 0xffffffff81008257
       46150       27589:           1 write        8 byte(s) @ 0xffff8800b904b0d0 by PC 0xffffffff81008257
       46151       27589:           1 write        8 byte(s) @ 0xffff8800b904b0d8 by PC 0xffffffff81008257
       46152       27589:           1 write        8 byte(s) @ 0xffff8800b904b0e0 by PC 0xffffffff81008257
       46153       27589:           1 write        8 byte(s) @ 0xffff8800b904b0e8 by PC 0xffffffff81008257
       46154       27589:           1 write        8 byte(s) @ 0xffff8800b904b0f0 by PC 0xffffffff81008257
       46155       27589:           1 write        8 byte(s) @ 0xffff8800b904b0f8 by PC 0xffffffff81008257
       46156       27589:           1 write        8 byte(s) @ 0xffff8800b904b100 by PC 0xffffffff81008257
       46157       27589:           1 write        8 byte(s) @ 0xffff8800b904b108 by PC 0xffffffff81008257
       46158       27589:           1 write        8 byte(s) @ 0xffff8800b904b110 by PC 0xffffffff81008257
       46159       27589:           1 write        8 byte(s) @ 0xffff8800b904b118 by PC 0xffffffff81008257
       46160       27589:           1 write        8 byte(s) @ 0xffff8800b904b120 by PC 0xffffffff81008257
       46161       27589:           1 write        8 byte(s) @ 0xffff8800b904b128 by PC 0xffffffff81008257
       46162       27589:           1 write        8 byte(s) @ 0xffff8800b904b130 by PC 0xffffffff81008257
       46163       27589:           1 write        8 byte(s) @ 0xffff8800b904b138 by PC 0xffffffff81008257
       46164       27589:           1 write        8 byte(s) @ 0xffff8800b904b140 by PC 0xffffffff81008257
       46165       27589:           1 write        8 byte(s) @ 0xffff8800b904b148 by PC 0xffffffff81008257
       46166       27589:           1 write        8 byte(s) @ 0xffff8800b904b150 by PC 0xffffffff81008257
       46167       27589:           1 write        8 byte(s) @ 0xffff8800b904b158 by PC 0xffffffff81008257
       46168       27589:           1 write        8 byte(s) @ 0xffff8800b904b200 by PC 0xffffffff81008257
       46169       27589:           1 write        8 byte(s) @ 0xffff8800b904b208 by PC 0xffffffff81008257
       46170       27589:           1 write        8 byte(s) @ 0xffff8800b904b210 by PC 0xffffffff81008257
       46171       27589:           1 write        8 byte(s) @ 0xffff8800b904b218 by PC 0xffffffff81008257
       46172       27589:           1 write        8 byte(s) @ 0xffff8800b904b220 by PC 0xffffffff81008257
       46173       27589:           1 write        8 byte(s) @ 0xffff8800b904b228 by PC 0xffffffff81008257
       46174       27589:           1 write        8 byte(s) @ 0xffff8800b904b230 by PC 0xffffffff81008257
       46175       27589:           1 write        8 byte(s) @ 0xffff8800b904b238 by PC 0xffffffff81008257
       46176       27589:           1 write        8 byte(s) @ 0xffff8800b904b240 by PC 0xffffffff81008257
       46177       27589:           1 write        8 byte(s) @ 0xffff8800b904b248 by PC 0xffffffff81008257
       46178       27589:           1 write        8 byte(s) @ 0xffff8800b904b250 by PC 0xffffffff81008257
       46179       27589:           1 write        8 byte(s) @ 0xffff8800b904b258 by PC 0xffffffff81008257
       46180       27589:           1 write        8 byte(s) @ 0xffff8800b904b260 by PC 0xffffffff81008257
       46181       27589:           1 write        8 byte(s) @ 0xffff8800b904b268 by PC 0xffffffff81008257
       46182       27589:           1 write        8 byte(s) @ 0xffff8800b904b270 by PC 0xffffffff81008257
       46183       27589:           1 write        8 byte(s) @ 0xffff8800b904b278 by PC 0xffffffff81008257
       46184       27589:           1 write        8 byte(s) @ 0xffff8800b904b280 by PC 0xffffffff81008257
       46185       27589:           1 write        8 byte(s) @ 0xffff8800b904b288 by PC 0xffffffff81008257
       46186       27589:           1 write        8 byte(s) @ 0xffff8800b904b290 by PC 0xffffffff81008257
       46187       27589:           1 write        8 byte(s) @ 0xffff8800b904b298 by PC 0xffffffff81008257
       46188       27589:           1 write        8 byte(s) @ 0xffff8800b904b2a0 by PC 0xffffffff81008257
       46189       27589:           1 write        8 byte(s) @ 0xffff8800b904b2a8 by PC 0xffffffff81008257
       46190       27589:           1 write        8 byte(s) @ 0xffff8800b904b2b0 by PC 0xffffffff81008257
       46191       27589:           1 write        8 byte(s) @ 0xffff8800b904b2b8 by PC 0xffffffff81008257
       46192       27589:           1 write        8 byte(s) @ 0xffff8800b904b2c0 by PC 0xffffffff81008257
       46193       27589:           1 write        8 byte(s) @ 0xffff8800b904b2c8 by PC 0xffffffff81008257
       46194       27589:           1 write        8 byte(s) @ 0xffff8800b904b2d0 by PC 0xffffffff81008257
       46195       27589:           1 write        8 byte(s) @ 0xffff8800b904b2d8 by PC 0xffffffff81008257
       46196       27589:           1 write        8 byte(s) @ 0xffff8800b904b2e0 by PC 0xffffffff81008257
       46197       27589:           1 write        8 byte(s) @ 0xffff8800b904b2e8 by PC 0xffffffff81008257

```

Reply:
Those contiguous ranges would be combined in DR into a single range instead of a million 8-byte writes: that seems silly to split every 8-byte into a separate record as even if xsaveopt generates separate memory transactions for different SIMD regs, etc. many of those are larger than 8 bytes and I'm sure it's not sending out separate 8-byte requests.